forgettable-frontend/src/pages/PersonPage/PersonPage.js

```diff
@@ -1,30 +1,106 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import EncounterCard from '../../components/EncounterCard/EncounterCard';
 import IconButton from '../../components/IconButton/IconButton';
 import PersonDrawer from '../../components/PersonDrawer/PersonDrawer';
+import {createPerson, getPerson} from '../../services';
 import classes from './PersonPage.module.css';
+import {ENCOUNTERS} from './PlaceholderData';
+import {useParams} from 'react-router-dom';
 
 const PersonPage = (props) => {
+  // TODO: this needs to be passed in as a routing parameter
+  const {id} = useParams();
+
+  const [person, setPerson] = useState({
+    firstName: '',
+    lastName: '',
+    birthday: null,
+    gender: '',
+    location: '',
+    howWeMet: '',
+    firstMet: null,
+    interests: [],
+    organisation: '',
+    socialMedia: [],
+    img: '',
+    encounters: [],
+    timeUpdated: null,
+  });
+
+  useEffect(async () => {
+    const result = await getPerson(id);
+
+    setPerson({
+      firstName: result.first_name,
+      lastName: result.last_name,
+      birthday: result.birthday,
+      gender: result.gender,
+      location: result.location,
+      howWeMet: result.how_we_met,
+      interests: result.interests,
+      organisation: result.organisation,
+      socialMedia: convertSocialMedia(result.socialMedia),
+      img: result.image,
+      encounters: result.encounters,
+      encounters: ENCOUNTERS,
+      timeUpdated: result.timeUpdated,
+    });
+  }, [id]);
+
+  const convertSocialMedia = (socialMedias) => {
+    if (!socialMedias) return null;
+
+    const socialMediaArray = [];
+
+    for (const [key, value] of Object.entries(socialMedias)) {
+      socialMediaArray.push({
+        name: key,
+        link: value,
+      });
+    }
+
+    return socialMediaArray;
+  };
+
+  const createEncounterComponent = (encounter, i) => {
+    return (
+      <div className={classes.CardWrapper} key={encounter.id || i}>
+        <EncounterCard
+          title={encounter.title}
+          description={encounter.description}
+          persons={encounter.persons}
+          location={encounter.location}
+          onClick={() => {}}
+          onDelete={() => {}}
+          date={encounter.date}
+          isInitialEncounter={false}
+        />
+      </div>
+    );
+  };
+
   return (
     <div className={classes.PersonPage}>
       <PersonDrawer
         open={true}
-        img="https://user-images.githubusercontent.com/62003343/158300159-495244dd-f05e-4384-9fac-ddf88a06be39.gif"
+        img={person.img}
```

Comment:
have you changed anything yet to make it support the image buffer that gets returned by the api? just asking cause im about to try figure that out haha

Reply:
Oooh I have not, didn't occur to me that that doesn't work haha I just thought the person didn't have a photo. Will include this in the PR to update the API call :D